website/docs/get-started-guide/dev-starter-guide/production-tips.md

```diff
@@ -0,0 +1,150 @@
+
+To do that, use the [shadow-loader](https://github.com/semrush/intergalactic/blob/master/tools/shadow-loader/README.md) webpack plugin. It will strip the styles from JavaScript and replace them with `require ("./style.css")` in the component code. This way you will be able to extract the styles into a separate file using [mini-css-extract-plugin](https://webpack.js.org/plugins/mini-css-extract-plugin/) or a similar tool.
+
+This is what your `webpack.config.js` might look like:
+
+```js
+const MiniCssExtractPlugin = require('mini-css-extract-plugin');
+module.exports = {
+  // ...
+  plugins: [new MiniCssExtractPlugin()],
+  module: {
+    rules: [
+      {
+        test: /\.js$/,
+        include: /node_modules\/(@semcore|\.cache\/reshadow)/,
+        use: '@semcore/shadow-loader',
+      },
+      {
+        test: /\.css$/,
+        include: /node_modules\/(@semcore|\.cache\/reshadow)/,
+        use: [MiniCssExtractPlugin.loader, 'css-loader'],
+      },
+    ],
+  },
+};
+```
+
+## Style isolation
+
+Classes in styles have a hash of the content and version to avoid collisions. However, this solution doesn't work when there are two components of the same version from two different teams on the same page. In such cases, styles may overlap and apply incorrectly due to issues with the order of the CSS cascade.
+
+To solve this problem, make the classes unique by using a prefix. Since we supply pre-built JavaScript and CSS files, there is no correct way to assign a prefix directly.
+
+The solution that we suggest is to replace the placeholder `_gg_` in the class names by using [string-replace-loader](https://www.npmjs.com/package/string-replace-loader).
+
+Note that the placeholder needs to be replaced in both JS and CSS files. This is what your `webpack.config.js` might look like:
+
+```js
+module.exports = {
+  // ...
+  module: {
+    rules: [
+      {
+        test: /\.(js|css)$/,
+        include: /node_modules\/(@semcore|\.cache\/reshadow)/,
+        enforce: 'pre',
+        use: [
+          {
+            loader: 'string-replace-loader',
+            options: {
+              search: '_gg_',
+              replace: '-my-team',
+              flags: 'g',
+            },
+          },
+        ],
+      },
+    ],
+  },
+};
+```
+
+Also note that the configuration of the `rule` parameter in JS files uses `enforce: 'pre'`. Otherwise, obfuscation will change the function names and make it impossible to search and replace CSS classes. Add these rules to your `webpack-config.js` as a separate section, since they should process packages from the @semcore library only.
+
+## Server-side rendering
+
+There are two ways to implement SSR.
+
+### SSR via sstyled.getStyles function
+
+This method will work for you if you're not using the @semcore/shadow-loader package (see [CSS extract](/internal/production/#a5c869)).
```

Comment:
Where should CSS extract link lead us? Cannot find the appropriate documentation.

Reply:
fixed